878 merge fill_desc and set_options

ilyaZar opened the pull request:
Fix #878.

Not sure (since it's been a while ..) if it is  __*still*__ planned to be a breaking change as mentioned in https://github.com/ThinkR-open/golem/issues/878#issuecomment-1212434319 .

It need not be a breaking change __*for the moment*__ so here is a "careful" version with a deprecation note for the user in the docs:

1. Add a new parameter `set_options = FALSE` to `fill_desc()`, so  `set_golem_options()` will not run per default. However, `set_golem_options()` is run when that flag is `TRUE` (see commit 1 to this PR).

2. That default `FALSE` is set inside files 01_start.R and a_start.Rmd for the user, adding a comment about future deprecation (see commit 2 to this PR).

3. For testing purposes one can `set_options = TRUE` in “build.R” (commit 3 to this PR) for some time.

ilyaZar committed:
refactor: add logical flag to run set_golem_options() inside fill_desc()

- defaults to FALSE for retrocompatibility
- add documentation informing about future deprecation

ilyaZar committed:
docs: inform user about changes to fill_desc()

- add a note that set_options flag will be TRUE in future golem version but keep the default to FALSE inside 01_start.R and corresponding vignette for the moment

ColinFay commented on R/desc.R:
```diff
@@ -30,7 +34,8 @@ fill_desc <- function(
   author_email = NULL,
-  author_orcid = NULL
+  author_orcid = NULL,
+  set_options = FALSE
```
I would love this to be set to TRUE by default :)

ilyaZar replied:
done, but I keep the versions in `01_start.R` and `a_start.Rmd` files to `FALSE`, or?

ColinFay replied:
No, let's put it to TRUE everywhere :) The idea with the merge of the two function is to be sure that this fun is run, so it's better if it's TRUE by default. Thanks !

ColinFay replied:
Changes commited, thanks :)

ilyaZar committed:
refactor: set_options defaults to TRUE

Since set_options=TRUE:

- remove manual setting of set_options=TRUE inside mantests/build.R
- update docs for fill_desc() this param

also: add missing one-liner description for role=c("aut", "cre") inside 01_start.R and a_start.Rmd which is consistent with the short descriptions of the other parameters in these files

ColinFay commented on inst/shinyexample/dev/01_start.R:
```diff
@@ -26,13 +26,17 @@ golem::fill_desc(
   repo_url = NULL, # The URL of the GitHub repo (optional),
-  pkg_version = "0.0.0.9000" # The version of the package containing the app
+  pkg_version = "0.0.0.9000", # The version of the package containing the app
+  set_options = FALSE # do not automatically run set_golem_options() but will
```
```suggestion
  set_options = TRUE # Set the options
```

ColinFay commented on inst/shinyexample/dev/01_start.R:
```diff
@@ -26,13 +26,17 @@ golem::fill_desc(
-  pkg_version = "0.0.0.9000" # The version of the package containing the app
+  pkg_version = "0.0.0.9000", # The version of the package containing the app
+  set_options = FALSE # do not automatically run set_golem_options() but will
+  # change to TRUE in the future, see below
```
```suggestion

```

ColinFay committed:
Update inst/shinyexample/dev/01_start.R

ColinFay committed:
Update inst/shinyexample/dev/01_start.R

ColinFay commented on vignettes/a_start.Rmd:
```diff
@@ -120,18 +120,21 @@ golem::fill_desc(
   repo_url = NULL, # The URL of the GitHub repo (optional),
-  pkg_version = "0.0.0.9000" # The version of the package containing the app
+  pkg_version = "0.0.0.9000", # The version of the package containing the app,
+  set_options = FALSE # do not automatically run set_golem_options() but will
```
```suggestion
  set_options = TRUE # Set the global golem options
```

ColinFay commented on vignettes/a_start.Rmd:
```diff
@@ -120,18 +120,21 @@ golem::fill_desc(
-  pkg_version = "0.0.0.9000" # The version of the package containing the app
+  pkg_version = "0.0.0.9000", # The version of the package containing the app,
+  set_options = FALSE # do not automatically run set_golem_options() but will
+  # change to TRUE in the future, see below
```
```suggestion

```

ColinFay committed:
Update vignettes/a_start.Rmd

ColinFay committed:
Update vignettes/a_start.Rmd
